internal/security/credentials_test.go

```diff
@@ -40,109 +38,203 @@ type getSecretsTestObj struct {
+		},
+
+		getSecretsTestObj{
+			testName:          "One valid key, one empty key",
+			path:              "db_secrets",
+			keys:              []string{"key1", ""},
+			expectedSecrets:   nil,
+			expectedErr:       errors.New("No value for the keys: [] exists"),
+			resetSecretsCache: true,
+		},
+		getSecretsTestObj{
+			testName:          "One valid key one not found key",
+			path:              "db_secrets",
+			keys:              []string{"key1", "notFoundKey"},
+			expectedSecrets:   nil,
+			expectedErr:       errors.New("No value for the keys: [notFoundKey] exists"),
+			resetSecretsCache: true,
+		},
+		getSecretsTestObj{
+			testName:          "Not found key",
+			path:              "db_secrets",
+			keys:              []string{"notFoundKey"},
+			expectedSecrets:   nil,
+			expectedErr:       errors.New("No value for the keys: [notFoundKey] exists"),
+			resetSecretsCache: true,
+		},
+		getSecretsTestObj{
+			testName:          "Two missing keys",
+			path:              "db_secrets",
+			keys:              []string{"notFoundKey1", "notFoundKey2"},
+			expectedSecrets:   nil,
+			expectedErr:       errors.New("No value for the keys: [notFoundKey1,notFoundKey2] exists"),
+			resetSecretsCache: true,
+		},
+		getSecretsTestObj{
+			testName:          "Valid key",
+			path:              "db_secrets",
+			keys:              []string{"key1"},
+			expectedSecrets:   map[string]string{"key1": "value1"},
+			expectedErr:       nil,
+			resetSecretsCache: true,
+		},
+		getSecretsTestObj{
+			testName:          "Two valid keys",
+			path:              "db_secrets",
+			keys:              []string{"key1", "key2"},
+			expectedSecrets:   map[string]string{"key1": "value1", "key2": "value2"},
+			expectedErr:       nil,
+			resetSecretsCache: true,
+		},
+		getSecretsTestObj{
+			testName:          "Valid key (key1 not cached)",
+			path:              "db_secrets",
+			keys:              []string{"key1"},
+			expectedSecrets:   map[string]string{"key1": "value1"},
+			expectedErr:       nil,
+			resetSecretsCache: false,
+		},
+		getSecretsTestObj{
+			testName:          "One valid key (key1 already cached)",
+			path:              "db_secrets",
+			keys:              []string{"key1"},
+			expectedSecrets:   map[string]string{"key1": "value1"},
+			expectedErr:       nil,
+			resetSecretsCache: false,
+		},
+		getSecretsTestObj{
+			testName:          "Two valid keys (key1 cached, key2 not cached)",
+			path:              "db_secrets",
+			keys:              []string{"key1", "key2"},
+			expectedSecrets:   map[string]string{"key1": "value1", "key2": "value2"},
+			expectedErr:       nil,
+			resetSecretsCache: false,
+		},
+	}
+
+	m.Run()
 }
 
 func TestGetSecrets(t *testing.T) {
 
-	secretProvider := newMockSecretProvider()
+	secretProvider := newMockSecretProvider(nil)
 
 	for i, test := range getSecretsTestData {
+		t.Run(test.testName, func(t *testing.T) {
+			secretProvider.secretClient.(*mockSecretClient).testIndex = i
```

Comment:
Suggest to using pinned local variable to avoid scopelint warning; i.e. make a copy of `test` and `i` to be used outside the t.Run().  Check this link https://www.reddit.com/r/golang/comments/64htjb/i_created_a_linter_tool_for_checking/ with https://play.golang.org/p/pfQtbVTf5q for potential issues if variables are not pinned.